app/presenters/tree_node_builder.rb

```diff
@@ -459,4 +463,63 @@ def miq_report_node(text, image, name, status)
       generic_node(text, image)
     end
   end
+
+  def assigned_server_role_node(object)
+    @node = {
+      :key      => build_object_id,
+      :addClass => "dynatree-title",
+      :title    => options[:tree] == :servers_by_role_tree ?
+        "#{Dictionary.gettext('MiqServer', :type => :model, :notfound => :titleize)}: #{object.name} [#{object.id}]" :
+        "Role: #{object.server_role.description}"
+    }
+
+    if object.master_supported?
+      priority = case object.priority
+                 when 1
+                   "primary, "
+                 when 2
+                   "secondary, "
+                 else
+                   ""
+                 end
+    end
+    if object.active? && object.miq_server.started?
+      @node[:icon] = ActionController::Base.helpers.image_path("100/on.png")
+      @node[:title] += _(" (%{priority}active, PID=%{number})") % {:priority => priority, :number => object.miq_server.pid}
```

Comment:
@h-kataria Hmm, perhaps we could move the whole gettext string to the case statement few
lines above:

```ruby
   if asr.master_supported?
      title = case asr.priority
                 when 1
                   _(" (primary, active, PID=%{number})")
                 when 2
                   _(" (secondary, active, PID=%{number})")
                 else
                   ""
                 end
    end
```

and then just

```ruby
node[:title] += title % {:number => asr.miq_server.pid}
```